### PR TITLE
allow disable highlight by default

### PR DESCRIPTION
**Pull request by ycrichard**
This is a quick fix to #2294. I also find this feature practically needed.

### CODE REVIEW

File: frontend/apps/reader/modules/readerhighlight.lua [hunk 1]
```diff
@@ -520,7 +522,13 @@ end
...
 function ReaderHighlight:onReadSettings(config)
     self.view.highlight.saved_drawer = config:readSetting("highlight_drawer") or self.view.highlight.saved_drawer
-    self.view.highlight.disabled = config:readSetting("highlight_disabled") or false
+    if config:readSetting("highlight_disabled")~=nil then
+        self.view.highlight.disabled = config:readSetting("highlight_disabled")
+    elseif G_reader_settings:readSetting("highlight_disabled")~=nil then
+        self.view.highlight.disabled = G_reader_settings:readSetting("highlight_disabled")
+    else
+        self.view.highlight.disabled = false
+    end
```
Comment (houqp) [inline frontend/apps/reader/modules/readerhighlight.lua]: Perhaps more efficient to write this block as below to reduce two function calls in the worst case ;)

```lua
local disable_hightlight = config:readSetting("highlight_disabled")
if disable_highlight == nil then
    disable_hightlight = G_reader_settings:readSetting("highlight_disabled") or false
end
self.view.highlight.disabled = disable_highlight
```

**Comment by houqp** [comment]
The rest of the patch looks good to me, thanks for adding this!

File: frontend/apps/reader/modules/readerhighlight.lua (hunk 1 above)
Comment (ycrichard) [inline frontend/apps/reader/modules/readerhighlight.lua]: I was wondering how to do it. This is better!

Comment (houqp) [inline frontend/apps/reader/modules/readerhighlight.lua]: OOPS, i made a typo in my code snippet (`hightlight`). Luckily the automated static analysis caught it. Could you fix it?

**Comment by houqp** [comment]
You should be able to get the same error with `make static-check` command.

**Comment by ycrichard** [comment]
ok, it's done.

**Comment by houqp** [comment]
Merged, thanks!

**Pull request merged**